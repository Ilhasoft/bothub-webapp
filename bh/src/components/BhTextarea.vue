<template>
  <bh-input>
    <div :class="inputClassAttr">
      <textarea
        ref="input"
        v-model="val"
        :class="`${className}__input`"
        v-bind="$attrs"
      />
    </div>
  </bh-input>
</template>

<script>
import BhText from './BhText';

export default {
  name: 'BhTextarea',
  extends: BhText,
  data() {
    return {
      className: 'bh-textarea',
    };
  },
  watch: {
    val() {
      this.updateTextareaHeight();
    },
  },
  async mounted() {
    await this.$nextTick();
    this.updateTextareaHeight();
  },
  methods: {
    updateTextareaHeight() {
      const { input } = this.$refs;
      const offset = input.offsetHeight - input.clientHeight;
      const computedStyle = window.getComputedStyle(input);
      input.style.height = computedStyle.getPropertyValue('min-height');
      input.style.height = `${offset + input.scrollHeight}px`;
    },
  },
};
</script>

<style lang="scss">
@import '@scss/forms.scss';


.bh {
  &-textarea {
    @include textarea();
  }
}
</style>
