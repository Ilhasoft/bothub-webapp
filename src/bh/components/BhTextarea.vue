<template>
  <bh-input>
    <div :class="inputClassAttr">
      <textarea
        ref="textarea"
        v-bind="$attrs"
        v-model="val" />
    </div>
  </bh-input>
</template>

<script>
import BhText from './BhText';

export default {
  name: 'BhTextarea',
  extends: BhText,
  async mounted() {
    await this.$nextTick();
    this.updateTextareaHeight();
  },
  watch: {
    val() {
      this.updateTextareaHeight();
    },
  },
  data() {
    return {
      className: 'bh-textarea',
    };
  },
  methods: {
    updateTextareaHeight() {
      const { textarea } = this.$refs;
      const offset = textarea.offsetHeight - textarea.clientHeight;
      const computedStyle = window.getComputedStyle(textarea);
      textarea.style.height = computedStyle.getPropertyValue('min-height');
      textarea.style.height = `${offset + textarea.scrollHeight}px`;
    },
  },
};
</script>

<style lang="scss">
@import '~bh/assets/scss/forms.scss';


.bh {
  &-textarea {
    @include textarea();
  }
}
</style>
