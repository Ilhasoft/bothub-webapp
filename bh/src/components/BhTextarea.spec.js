/* eslint-env jest */
import { shallowMount, createLocalVue } from '@vue/test-utils';

import BH from '..';
import BhTextarea from './BhTextarea';


const localVue = createLocalVue();
localVue.use(BH);

describe('BhTextarea.vue', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallowMount(BhTextarea, {
      localVue,
    });
  });

  test('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
