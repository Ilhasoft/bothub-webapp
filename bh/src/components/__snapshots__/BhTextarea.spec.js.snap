// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BhTextarea.vue renders correctly 1`] = `
VueWrapper {
  "_emitted": Object {},
  "_emittedByOrder": Array [],
  "isFunctionalComponent": undefined,
}
`;
